### DISCUSSION TITLE
In the Setup Wizard, choose whether to associate files

### DISCUSSION THREAD

**Discussion by shiumano** [ppy/osu#29603] [Ideas] [answered]
Many people say “I double-clicked beatmap and it opened lazer instead of stable, so I uninstalled it.”
It is not a good idea to associate files with lazer when stable is currently more dominant.
It would be nice to be able to choose whether or not to associate files during initial setup.

**Comment by smallketchup82** [comment]
I'm currently working on moving lazer's update framework to Velopack, and I've been backlogging some discussions and issues with the installer to see if there's anything I can fix, and found this discussion post.

To put it shortly, while this *can* be done, it'll have some caveats. Probably won't be possible to put it into the installer, at least not until Velopack adds NSIS/WiX support. Could probably add this to the setup wizard, but I don't believe the core team wants any more settings than necessary.

I had planned to revisit associations & clean them up post-Velopack, but I might not have the time for it. Velopack itself looks as though it might support file associations some day. Regardless, expect file associations to work a bit better after the move to Velopack. And if it gets refactored, even better. But also note that the chances of there ever being a setting for this are unlikely (especially as lazer is beginning to gain more adoption).

**Reply by shiumano** [reply]
Well, well...this suggestion is a temporary measure in case people install Lazer while Stable is still in its heyday.
Of course, once Lazer becomes the default, the files should be associated, so i'm talking about adding a code that is only relevant now if you don't mind adding it.

**Reply by catuhana** [reply]
I don't know if that's the planned way or if there's even a plan, but wouldn't using Windows' Default Apps settings (I don't know the API for it) be better for both Lazer and stable? This way, all osu file extension can be registered to Windows, and osu!lazer and stable would be the apps that could be run around those file extensions, resulting on a screen like this:
![image](https://github.com/user-attachments/assets/7e85c6f7-61f0-499d-abef-cff4691941f3)

And that's how Minecraft's file associations look like (osu!lazer does not have that):
![image](https://github.com/user-attachments/assets/c35205bc-17e8-4e84-84ea-e51e86837ee7)

This way, the user can just choose which one to open, which is way better than clashing default apps, I think.

The only issue with this is probably stable release would need a rewrite for this feature too.

**Reply by smallketchup82** [reply]
This is already what lazer does

**Comment by Susko3** [comment, marked as answer]
https://github.com/ppy/osu/issues/30473
